core/src/main/scala/akka/kafka/ConsumerSettings.scala

```diff
@@ -24,12 +24,31 @@ sealed trait ManualSubscription extends Subscription
 sealed trait AutoSubscription extends Subscription
 
 object Subscriptions {
+  private[kafka] final case class TopicSubscriptionWithStartTimestamp(timestamp: Long, tps: Set[String]) extends AutoSubscription
   private[kafka] final case class TopicSubscription(tps: Set[String]) extends AutoSubscription
   private[kafka] final case class TopicSubscriptionPattern(pattern: String) extends AutoSubscription
   private[kafka] final case class Assignment(tps: Set[TopicPartition]) extends ManualSubscription
   private[kafka] final case class AssignmentWithOffset(tps: Map[TopicPartition, Long]) extends ManualSubscription
   private[kafka] final case class AssignmentOffsetsForTimes(timestampsToSearch: Map[TopicPartition, Long]) extends ManualSubscription
 
+  /**
+   * Creates subscription starting from `timestamp` for given set of topics
+   */
+  def topicsWithTimestamp(timestamp: Long, ts: Set[String]): AutoSubscription = TopicSubscriptionWithStartTimestamp(timestamp, ts)
```

Comment:
Perhaps `topicsFromTimestamp` or `topicsWithStartTimestamp`?

Reply:
I like the latter the most, renamed it.